tsconfig.dev.json

```diff
@@ -0,0 +1,5 @@
+{
+  "extends": "./tsconfig",
+  "target": "esnext",
+  "noUnusedLocals": false
```

Comment:
```suggestion
```

Sorry, but this is something I think adds value.

Reply:
It is disabled only in development mode, but it is enabled in production.

Current DX:
- I want to test something, I comment out some code
- TS compiler complains about unused variable
- I comment out more code
- TS compiler complains about more unused variable and maybe imports
- I need to comment out more code

Taking into account that TS compilations are slow (at least 1.5 s) this is very painful experience

Reply:
I still think it's valuable because it enforces clean code. In my opinion, the solution in this case is not to remove the safety, but to [increase the speed](https://devblogs.microsoft.com/typescript/announcing-typescript-3-4-rc/).

Reply:
I think it can be very valuable, it's true that it's a bit painful sometime 😅 My only concern is: can we make sure that we are not committing unused variable (so part of commit hook) if it's the case, it's good to me 😃

Reply:
It should lint warning, not a compiler error in development. And compiler error in production. We can run linter with different config in precommit hook, but on the other side we can also just fail CI if there are unused variables.

tslint AFAIK deprecated this rule. We can switch from tslint to eslint (as soon as they will support TS) and eslint have this rule.

Reply:
What if we add a tsconfig.precommit.json and use it in the hook with maximum strictness?

The thing is, right now, we don’t run tsc at all on the book because it’s not as straightforward as it seems, but this could be a solution. 

As far as I’m concerned though — this check **has to execute somewhere** because I’ve seen a lot of problems without it in past experience.